Interface.write: java.net.SocketException: Software caused connection abort: socket write error

Everytime i try to run the original code that comes with the main file 

`S7Connector c = 
				S7ConnectorFactory
				.buildTCPConnector()
				.withHost("192.168.0.1")
				.build();
		
		S7Serializer s = S7SerializerFactory.buildSerializer(c);
		
		DB db = new DB();
		db.str = "Hello!";
		db.d = Math.PI;
		db.b1 = true;
		db.b3 = true;
		db.by1 = 0x5A;
		
		Date d = new Date();
		System.out.println(d);
		System.out.println(d.getTime());
		
		db.date1 = d;
		db.date2 = d;
		db.millis = 3600000;
		
		s.store(db, 100, 0);

		System.out.println("OK");
		
		c.close();`
i get this 

`Interface.write: java.net.SocketException: Software caused connection abort: socket write error
10:09:24.850 [main] TRACE c.g.s.i.serializer.parser.BeanParser - Parsing: com.github.s7connector.test.live.BigDBTest$DB
10:09:24.865 [main] TRACE c.g.s.i.serializer.parser.BeanParser - Parsing field: bytes
10:09:24.866 [main] TRACE c.g.s.i.serializer.parser.BeanParser - 		type: BYTE
10:09:24.866 [main] TRACE c.g.s.i.serializer.parser.BeanParser - 		byteOffset: 0
10:09:24.866 [main] TRACE c.g.s.i.serializer.parser.BeanParser - 		bitOffset: 0
10:09:24.866 [main] TRACE c.g.s.i.serializer.parser.BeanParser - 		size: 0
10:09:24.866 [main] TRACE c.g.s.i.serializer.parser.BeanParser - 		arraySize: 4096
10:09:24.866 [main] TRACE c.g.s.i.serializer.parser.BeanParser - 	New blocksize (+offset): 0
10:09:24.866 [main] TRACE c.g.s.i.serializer.parser.BeanParser - 	New blocksize (+array): 4096
10:09:24.866 [main] TRACE c.g.s.i.serializer.parser.BeanParser - Parsing done, overall size: 4096
10:09:24.866 [main] TRACE c.g.s.i.serializer.S7SerializerImpl - store-buffer-size: 4096
10:09:24.867 [main] TRACE c.g.s.i.serializer.S7SerializerImpl - Inerting buffer with size: 4096 at offset 0 into bean: com.github.s7connector.test.live.BigDBTest$DB@33c7e1bb
10:09:24.868 [main] TRACE c.g.s.i.serializer.parser.BeanParser - Parsing: com.github.s7connector.test.live.BigDBTest$DB
10:09:24.868 [main] TRACE c.g.s.i.serializer.parser.BeanParser - Parsing field: bytes
10:09:24.868 [main] TRACE c.g.s.i.serializer.parser.BeanParser - 		type: BYTE
10:09:24.869 [main] TRACE c.g.s.i.serializer.parser.BeanParser - 		byteOffset: 0
10:09:24.869 [main] TRACE c.g.s.i.serializer.parser.BeanParser - 		bitOffset: 0
10:09:24.869 [main] TRACE c.g.s.i.serializer.parser.BeanParser - 		size: 0
10:09:24.869 [main] TRACE c.g.s.i.serializer.parser.BeanParser - 		arraySize: 4096
10:09:24.869 [main] TRACE c.g.s.i.serializer.parser.BeanParser - 	New blocksize (+offset): 0
10:09:24.869 [main] TRACE c.g.s.i.serializer.parser.BeanParser - 	New blocksize (+array): 4096
10:09:24.869 [main] TRACE c.g.s.i.serializer.parser.BeanParser - Parsing done, overall size: 4096
Interface.write: java.net.SocketException: Software caused connection abort: socket write error
Exception in thread "main" com.github.s7connector.exception.S7Exception: store
	at com.github.s7connector.impl.serializer.S7SerializerImpl.store(S7SerializerImpl.java:187)
	at com.github.s7connector.test.live.BigDBTest.main(BigDBTest.java:59)
Caused by: java.lang.IllegalArgumentException: Result: no message defined for code: 4096!
	at com.github.s7connector.impl.S7BaseConnection.checkResult(S7BaseConnection.java:55)
	at com.github.s7connector.impl.S7BaseConnection.write(S7BaseConnection.java:124)
	at com.github.s7connector.impl.S7BaseConnection.write(S7BaseConnection.java:118)
	at com.github.s7connector.impl.serializer.S7SerializerImpl.store(S7SerializerImpl.java:185)
	... 1 more`

--- a/src/main/java/com/github/s7connector/impl/nodave/TCPConnection.java
+++ b/src/main/java/com/github/s7connector/impl/nodave/TCPConnection.java
@@ -57,24 +57,24 @@ public final class TCPConnection extends S7Connection {
      * @return the int
      */
     public int connectPLC() {
-        final byte[] b4 = {
-                (byte) 0x11, (byte) 0xE0, (byte) 0x00, (byte) 0x00, (byte) 0x00, (byte) 0x01, (byte) 0x00,
-                (byte) 0xC1, (byte) 0x02, (byte) 0x01, (byte) 0x00, (byte) 0xC2, (byte) 0x02, (byte) 0x01, (byte) 0x02,
-                (byte) 0xC0, (byte) 0x01, (byte) 0x09
-        };
-        final byte[] b243 = {
-                (byte) 0x11, (byte) 0xE0, (byte) 0x00, (byte) 0x00, (byte) 0x00, (byte) 0x01, (byte) 0x00,
-                (byte) 0xC1, (byte) 0x02, (byte) 0x4D, (byte) 0x57, (byte) 0xC2, (byte) 0x02, (byte) 0x4D, (byte) 0x57,
-                (byte) 0xC0, (byte) 0x01, (byte) 0x09
-        };
         int packetLength;
         if (iface.protocol == Nodave.PROTOCOL_ISOTCP243) {
+        	final byte[] b243 = {
+        			(byte) 0x11, (byte) 0xE0, (byte) 0x00, (byte) 0x00, (byte) 0x00, (byte) 0x01, (byte) 0x00,
+        			(byte) 0xC1, (byte) 0x02, (byte) 0x4D, (byte) 0x57, (byte) 0xC2, (byte) 0x02, (byte) 0x4D, (byte) 0x57,
+        			(byte) 0xC0, (byte) 0x01, (byte) 0x09
+        	};
             System.arraycopy(b243, 0, this.msgOut, 4, b243.length);
             packetLength = b243.length;
         } else {
+        	final byte[] b4 = {
+        			(byte) 0x11, (byte) 0xE0, (byte) 0x00, (byte) 0x00, (byte) 0x00, (byte) 0x01, (byte) 0x00,
+        			(byte) 0xC1, (byte) 0x02, (byte) 0x01, (byte) 0x00, (byte) 0xC2, (byte) 0x02, (byte) 0x01, (byte) 0x02,
+        			(byte) 0xC0, (byte) 0x01, (byte) 0x09
+        	};
+        	System.arraycopy(b4, 0, this.msgOut, 4, b4.length);
             this.msgOut[17] = (byte) (this.rack + 1);
             this.msgOut[18] = (byte) this.slot;
-            System.arraycopy(b4, 0, this.msgOut, 4, b4.length);
             packetLength = b4.length;
         }
         this.sendISOPacket(packetLength);

